addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -13322,4 +13322,75 @@ QUnit.module("Views", (hooks) => {
             await click(target, ".o_form_button_save");
         }
     );
+
+    QUnit.only("multiple views for m2m field after list item edit in form", async function (assert) {
+
+        serverData.models.foo = {
+            fields: {
+                m2m: { string: "M2M field", type: "many2many", relation: "bar" },
+            },
+            records: [
+                {
+                    id: 1,
+                    m2m: [1, 2]
+                }
+            ]
+        }
+
+        serverData.models.bar = {
+            fields: {
+                m2m: { string: "M2M field", type: "many2many", relation: "extra" },
+            },
+            records: [
+                { id: 1, display_name: "ma", m2m: [1] },
+                { id: 2, display_name: "cr", m2m: [2] },
+            ],
+        };
+
+        serverData.models.extra = {
+            fields: {
+                name: { string: "Char", type: "char" },
+            },
+            records: [
+                { id: 1, display_name: "ma", name: "ma" },
+                { id: 2, display_name: "cr", name: "cr" },
+            ],
+        };
```

Comment:
Why don t use the default serverData ?

Reply:
I changed it to make use more of default serverData , but had to change some models to get the intended scenario.